### PR TITLE
Update doc index

### PR DESCRIPTION
**Pull request by lhoestq**

### CODE REVIEW

File: stream_opt_in_out.py
Comment (AndreaFrancis) [inline stream_opt_in_out.py:1]: Is this file needed for the PR?

Comment (lhoestq) [reply stream_opt_in_out.py:1]: whoops I forgot to remove it from my PR sorry ^^

File: docs/source/index.mdx [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
 
-Datasets Server is a lightweight web API for visualizing and exploring all types of datasets - computer vision, speech, text, and tabular - stored on the Hugging Face [Hub](https://huggingface.co/datasets). As datasets increase in size and data type richness, the cost of preprocessing (storage and compute) these datasets can be challenging and time-consuming. To help users access these modern datasets, Datasets Server runs a server behind the scenes to generate the API responses ahead of time and stores it in a database so they are instantly returned when you make a query through the API. 
+Datasets Server is a lightweight web API for visualizing and exploring all types of datasets - computer vision, speech, text, and tabular - stored on the Hugging Face [Hub](https://huggingface.co/datasets).
+As datasets increase in size and data type richness, the cost of preprocessing (storage and compute) these datasets can be challenging and time-consuming.
+To help users access these modern datasets, Datasets Server runs a server behind the scenes to generate the API responses ahead of time and stores it in a database so they are instantly returned when you make a query through the API.
```
Comment (AndreaFrancis) [inline docs/source/index.mdx:5]:
```suggestion
To help users access these modern datasets, Datasets Server runs a server behind the scenes to generate the API responses ahead of time and stores them in a database so they are instantly returned when you make a query through the API. 
```

**Review by AndreaFrancis — APPROVED** [review]

**Review by severo — APPROVED** [review]
